lib/slather/project.rb

```diff
@@ -79,12 +80,19 @@ def derived_data_path
         schemeArgument = "-scheme \"#{self.scheme}\""
         buildAction = "test"
       else
-        schemeArgument = nil
-        buildAction = nil
+        schemeArgument = ''
+        buildAction = ''
       end
 
-      # redirect stderr to avoid xcodebuild errors being printed.
-      build_settings = `xcodebuild #{projectOrWorkspaceArgument} #{schemeArgument} -showBuildSettings #{buildAction} 2>&1`
+      build_settings = ''
+      Open3.popen2e('xcodebuild', projectOrWorkspaceArgument, schemeArgument, '-showbuildsettings', buildAction) do |stdin, stdout_err, wait_thr|
```

Comment:
At least with Xcode 8 I'm getting an error when not specifying the argument in camel case, like `-showBuildSettings`